### ISSUE TITLE
Environment variables not propagating into recipe inside Makefile

### ISSUE THREAD

**Issue by jamesbraza** [mingw-w64/mingw-w64#16]
I have the following `Makefile` that works fine on macOS (Monterey 12.6) with `gcc` (clang 14.0.0), and on Windows (10 Pro) with Cygwin (`3.3.6-341.x86_64`).

```Makefile
SHELL=/bin/bash -euo pipefail

REPO_ROOT = $(shell pwd)
export VIRTUAL_ENV := ${REPO_ROOT}/venv
# bin = POSIX, Scripts = Windows
export PATH := ${VIRTUAL_ENV}/bin:${VIRTUAL_ENV}/Scripts:${PATH}

show-python:    ## Show path to python and version
    @echo -n "python location: "
    @python -c "import sys; print(sys.executable, end='')"
    @echo -n ", version: "
    @python -c "import platform; print(platform.python_version())"

install: show-python
install:    ## Install all dev dependencies into a local virtual environment.
    python -m pip install -r requirements-dev.txt --progress-bar off
```

However, with MINGW64 from Git `2.38.1.windows.1` (MINGW64 version is from [`msys2-runtime 3.3.6-1`](https://github.com/git-for-windows/build-extra/blob/0b25881dffc7d1bd27d0138af8614b1dca969d6c/versions/package-versions-2.38.1-MinGit.txt#L48)), it doesn't work.  The `install` recipe requires a pass through of the `VIRTUAL_ENV` environment variable using `;`:

```Makefile
install: show-python
install:    ## Install all dev dependencies into a local virtual environment.
    export VIRTUAL_ENV="${VIRTUAL_ENV}"; \
    python -m pip install -r requirements-dev.txt --progress-bar off
```

Why is that?  I believe MINGW64 is lacking some `Makefile` compatibility here.

**Comment by lhmouse** [comment]
Are you attempting to invoke the `make` from _Git for Windows_? Are you invoking `make` or `mingw32-make`? Which directory does it belong in?

**Comment by jamesbraza** [comment]
I am using Git Bash to invoke `make`.  The `make` command is [`make` version 4.3 installed via chocolatey](https://community.chocolatey.org/packages/make).

```bash
HOSTNAME+user@hostname MINGW64 /
$ which make
/c/ProgramData/chocolatey/bin/make
```

Does that answer your questions?

**Comment by lhmouse** [comment]
Can you reproduce this issue with MSYS2 `make`? I suspect this behavior is caused by chocolatey's make itself, or by some other issues in communication between MSYS2 bash and non-MSYS2 programs.

**Comment by jamesbraza** [comment]
Okay, so I did the following:

```bash
choco install msys2
```

Then I opened `C:\tools\msys64\msys2.exe` and ran:

```bash
pacman -S git make
cd /c/path/to/repo
make
```

And it worked!  So this confirms Git Bash using MINGW64 is indeed having some weird interaction with the chocolatey-installed `make`.  Thanks for helping root cause this @lhmouse!

---

I still think this is a bug with MINGW64, as it's having a silent internal failure when working with `make` installed via cholcolatey (as suggested here: https://stackoverflow.com/a/59268508).  What is the next step?

**Comment by lhmouse** [comment]
No, don't use it as a development environment. The shell from Git for Windows is for use with Git only. If you need a 'sane' shell for development , use MSYS2 instead.

**Comment by jamesbraza** [comment]
Okay sounds good, thank you!

If the conclusion is just "don't use Git Bash" for `Makefile`s, then I think it's fine to close this one out.
